processor/src/services/adyen-payment.service.ts

```diff
@@ -397,11 +405,20 @@ export class AdyenPaymentService extends AbstractPaymentService {
   }
 
   async capturePayment(request: CapturePaymentRequest): Promise<PaymentProviderModificationResponse> {
+    // Only process if the payment method requires it so we don't fetch order/cart unnecessary.
+    let adyenLineItems: LineItem[] | undefined = undefined;
+    if (
+      request.payment.paymentMethodInfo.method &&
+      METHODS_REQUIRE_LINE_ITEMS.includes(request.payment.paymentMethodInfo.method)
+    ) {
+      adyenLineItems = await this.retrievePaymentAssociatedLineItems(request.payment.id);
+    }
```

Comment:
There are multiple ways of doing the logic and retrieving the order/cart and then extracting the Adyen line items from it. I chose this way cause the `AdyenApi` supports sending either a `LineItem[] | undefined` for the `lineItem` property in the payload to the `captureAuthorisedPayment` via the converter.

This way the properties line up nicely.